pkg/machines/vmnetworktab.jsx

```diff
@@ -0,0 +1,136 @@
+/*jshint esversion: 6 */
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import cockpit from 'cockpit';
+import { changeNetworkState } from "./actions.es6";
+import { Listing, ListingRow } from 'cockpit-components-listing.jsx';
+
+const _ = cockpit.gettext;
+
+const VmNetworkTab = ({ vm, dispatch }) => {
+    if (!vm.interfaces || vm.interfaces.length === 0) {
+        return (<div>{_("No network interfaces defined for this VM")}</div>);
+    }
+
+    const onChangeState = (network) => {
+        return () => {
+            if (network.mac) {
+                dispatch(changeNetworkState(vm, network.mac, network.state === 'up' ? 'down' : 'up'));
+            }
+        }
+    }
+    const addressPortSource = (source) => (<table>
+        <tr><td className='machines-network-source-descr'>{_("Address:")}</td><td className='machines-network-source-value'>{source.address}</td></tr>
+        <tr><td className='machines-network-source-descr'>{_("Port:")}</td><td className='machines-network-source-value'>{source.port}</td></tr>
+    </table>);
+
+    // Network data mapping to rows
+    const detailMap = [
+        { name: _("Type: "), value: 'type', header: true },
+        { name: _("Model type:"), value: 'model' },
+        { name: _("MAC Address:"), value: 'mac' },
+        { name: _("Target:"), value: 'target' },
+        { name: _("Source:"), value: (network) => {
+            const mapSource = {
+                direct: (source) => source.dev,
+                network: (source) => source.network,
+                bridge: (source) => source.bridge,
+                mcast: addressPortSource,
+                server: addressPortSource,
+                client: addressPortSource,
+                udp: addressPortSource,
+            }
+            if (mapSource[network.type] !== undefined) {
+                return mapSource[network.type](network.source);
+            } else {
+                return null;
+            }
+        }},
+        { name: _("Additional:"), value: (network) => {
+            const additionalMap = [
+                { name: _("MTU:"), value: 'mtu' },
+                { name: _("Virtualport:"), value: 'virtualportType' },
+                { name: _("Managed:"), value: 'managed' },
```

Comment:
All values like that must be translated.
Please identify all of them and use dictionary-based translation within helpers.es6:rephraseUI()